backend/src/ml_space_lambda/data_access_objects/app_configuration.py

```diff
@@ -0,0 +1,382 @@
+#
+#   Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+#   Licensed under the Apache License, Version 2.0 (the "License").
+#   You may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+
+from __future__ import annotations
+
+import json
+from typing import Dict, List, Optional
+from venv import logger
+
+from dynamodb_json import json_util as dynamodb_json
+
+from ml_space_lambda.data_access_objects.dynamo_data_store import DynamoDBObjectStore
+from ml_space_lambda.enums import ResourceType
+from ml_space_lambda.utils.mlspace_config import get_environment_variables
+
+
+class AppConfigurationModel:
+    def __init__(
+        self,
+        configScope: str,  # 'global' or a project name
+        version_id: int,
+        configuration: SettingsModel,
+        changed_by: str,
+        change_reason: str,
+    ):
+        self.configScope = configScope
+        self.version_id = version_id
+        self.configuration = configuration
+        self.changed_by = changed_by
+        self.change_reason = change_reason
+
+    def to_dict(self) -> dict:
+        return {
+            "configScope": self.configScope,
+            "versionId": self.version_id,
+            "configuration": self.configuration.to_dict(),
+            "changedBy": self.changed_by,
+            "changeReason": self.change_reason,
+        }
+
+    @staticmethod
+    def from_dict(dict_object: dict) -> AppConfigurationModel:
+        return AppConfigurationModel(
+            dict_object["configScope"],
+            dict_object["versionId"],
+            dict_object["configuration"],
```

Comment:
This should be `SettingsModel.from_dict(dict_object["configuration"]),`, changing in next commit